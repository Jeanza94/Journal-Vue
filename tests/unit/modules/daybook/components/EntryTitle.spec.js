import { shallowMount } from "@vue/test-utils";
import EntryTitle from '@/modules/daybook/components/EntryTitle'
import { journalState } from '../../../mock-data/test-journal-state';

describe('Pruebas en el componente EntryTitle', () => { 

    const mockRouter = { push: jest.fn() }

    const wrapper = shallowMount(EntryTitle, {
        props: {
            entry: {
                ...journalState.entries[0]
            }
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })
  
    test('debe de hacer match con el snapshot', () => { 
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('debe de redireccionar al hacer click en el entry-container', () => {
        wrapper.find('div').trigger('click');
        expect(mockRouter.push).toHaveBeenCalledTimes(1);
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'entry', params: { id: journalState.entries[0].id } });
    });

    test('Pruebas en las propiedades computadas', () => { 

        expect(wrapper.vm.day).toBe(6);
        expect(wrapper.vm.month).toBe('Diciembre');
        expect(wrapper.vm.yearDay).toBe('2022, Martes');
        expect(wrapper.vm.shortText).toBe(journalState.entries[0].text);

    });

});