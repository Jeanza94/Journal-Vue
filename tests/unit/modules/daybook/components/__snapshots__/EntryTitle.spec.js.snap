// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente EntryTitle debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">6</span><span class="mx-1 fs-5">Diciembre</span><span class="mx-2 fw-light">2022, Martes</span></div>
  <div class="entry-description">subir una foto</div>
</div>
`;
